src/components/Form/Radiobutton.tsx

```diff
@@ -0,0 +1,156 @@
+import React, { Component, ReactNode } from 'react';
+import classnames from 'classnames';
+import { HtmlInput, HtmlLabel, HtmlSpan, HtmlDiv } from '../../reset';
+import { idGenerator } from '../../utils/uuid';
+import { logger } from '../../utils/logger';
+import {
+  RadiobuttonGroupConsumer,
+  RadiobuttonGroupProviderState,
+} from './RadiobuttonGroup';
+
+const baseClassName = 'fi-radiobutton';
+
+const radiobuttonClassNames = {
+  container: `${baseClassName}_container`,
+  input: `${baseClassName}_input`,
+  label: `${baseClassName}_label`,
+  hintText: `${baseClassName}_hintText`,
+};
+
+type RadiobuttonVariant = 'small' | 'large';
+
+export interface RadiobuttonProps {
+  /** Custom classname to extend or customize */
+  className?: string;
+  /**
+   * Label element content
+   */
+  children?: ReactNode;
+  /** Group multiple Radiobuttons to be inside same selection group */
+  name?: string;
+  /** Unique value to be used inside Radiobutton group for this item */
+  value?: string;
+  /**
+   * Unique id
+   * If no id is specified, one will be generated using uuid
+   * @default uuidV4
+   */
+  id?: string;
+  /**
+   * Hint text to be displayed under the label.
+   */
+  hintText?: string;
+  /** Disable Radiobutton. Value won't be included when submitting */
+  disabled?: boolean;
+  /**
+   * 'small' | 'large'
+   * @default small
+   */
+  variant?: RadiobuttonVariant;
+  radiobuttonGroup?: boolean;
+  consumer?: RadiobuttonGroupProviderState;
+  checked?: boolean;
+  onChange?: (event: React.ChangeEvent<HTMLInputElement>) => void;
+}
+
+export interface RadiobuttonState {
+  checkedState: boolean;
+}
+
+class RadiobuttonItem extends Component<RadiobuttonProps> {
+  state = {
+    checkedState: !!this.props.checked,
```

Comment:
What happens if user adds shorthand like this: `<RadiobuttonItem checked ...`? Just wondering if this truthy check is sufficient or should it actually check if the property exists in props like we had to do for the value for dropdown?

Reply:
Seems to work as expected.